### ISSUE TITLE
allow admins to archive users

### ISSUE THREAD

**Issue by tmfrnz** [marine-defrag/marine-defrag-server#103]
- [x] add `is_archive` boolean or `archived_at` date field to users table and API
- [x] allow admin users to update "archive" field above
- [x] preventing archived users from logging in
- [x] strip archived users of all privileges (in case they still have an active session)

**Comment by parndt** [comment]
> strip archived users of all privileges (in case they still have an active session)

@tmfrnz what about if we just expire their token?

**Comment by tmfrnz** [comment]
> > strip archived users of all privileges (in case they still have an active session)
> 
> @tmfrnz what about if we just expire their token?

yes great idea, I guess that should be sufficient @parndt

**Comment by tmfrnz** [comment]
hi @parndt standard endpoint should be fine, thanks

**Comment by parndt** [comment]
@tmfrnz do you need `archived_at` or `archived` or `active` or something sent in the regular GET user payload at `/users/:id`?

**Comment by tmfrnz** [comment]
> @tmfrnz do you need `archived_at` or `archived` or `active` or something sent in the regular GET user payload at `/users/:id`?

@parndt for GET my preference would be to have both `is_archived` and `archived_at` - for the PUT, I would prefer to only have `is_archived` and let the server set the `archived_at` date

**Comment by tmfrnz** [comment]
PUT request with `is_archived: true` (or `is_archived: "true"`) does not appear to work

**Comment by tmfrnz** [comment]
@parndt could you have another look? here is the incoming PUT request

```
INFO -- : [...] Processing by UsersController#update as JSON
INFO -- : [...]   Parameters: {"created_at"=>"2022-03-14T18:49:38+01:00", "updated_at"=>"2024-05-15T14:46:51+02:00", "updated_by_id"=>nil, "email"=>"another@test.de", "name"=>"another test", "archived_at"=>nil, "is_archived"=>"true", "id"=>"6", "user"=>{"email"=>"another@test.de", "created_at"=>"2022-03-14T18:49:38+01:00", "updated_at"=>"2024-05-15T14:46:51+02:00", "name"=>"another test", "updated_by_id"=>nil, "archived_at"=>nil}}
```

**Comment by parndt** [comment]
> INFO -- : [...]   Parameters: {"created_at"=>"2022-03-14T18:49:38+01:00", "updated_at"=>"2024-05-15T14:46:51+02:00", "updated_by_id"=>nil, "email"=>"another@test.de", "name"=>"another test", "archived_at"=>nil, "is_archived"=>"true", "id"=>"6", "user"=>{"email"=>"another@test.de", "created_at"=>"2022-03-14T18:49:38+01:00", "updated_at"=>"2024-05-15T14:46:51+02:00", "name"=>"another test", "updated_by_id"=>nil, "archived_at"=>nil}}

Is this being done by an admin user @tmfrnz ? Is `archived_at` also being passed?

**Comment by dumpark** [comment]
yes (admin user), and no (only passing is_archived, assuming server will
set archived_at)

**Comment by parndt** [comment]
Closed by #110
